wrapper.py

```diff
@@ -1,84 +1,89 @@
+from utils import timed
+
+
 class Kalah:
-    def __init__(self):
+    __tup = (0, 0, 0, 0, 0, 0)
+    __p1r = slice(0, 6)
+    __p2r = slice(7, 13)
+    __r = range(14)
+    __h = 7
+    __depth = 5
+
+    def __init__(self, funcs):
         self.current_state = [6, 6, 6, 6, 6, 6, 0, 6, 6, 6, 6, 6, 6, 0]
-        self.player = 1
+        self.player = 0
+        self.fs = funcs
 
     def is_end(self):
-        if self.current_state[:6].count(0) == 6:
-            for i in range (7, 13):
-                self.current_state[13] += self.current_state[i]
-                self.current_state[i] = 0
+        if not sum(self.current_state[self.__p1r]):
+            self.current_state[13] = 72 - self.current_state[6]
+            self.current_state[self.__p2r] = self.__tup
             return True
-        elif self.current_state[7:13].count(0) == 6:
-            for i in range (0, 6):
-                self.current_state[6] += self.current_state[i]
-                self.current_state[i] = 0
+        elif not sum(self.current_state[self.__p2r]):
+            self.current_state[6] = 72 - self.current_state[13]
+            self.current_state[self.__p1r] = self.__tup
             return True
-
         return False
 
+    def _transfer_stones(self, pos, ign=13):
+
+        num_stones, self.current_state[pos] = self.current_state[pos], 0
+        pos += 1
+
+        if ign == 13:
+            for _ in range(num_stones):
+                if pos == 13:
+                    pos = 0
+                self.current_state[pos], pos = self.current_state[pos] + 1, pos + 1
+        else:
+            for _ in range(num_stones):
+                if pos == 6:
+                    pos = 7
+                if pos == 14:
+                    pos = 0
+                self.current_state[pos], pos = self.current_state[pos] + 1, pos + 1
```

Comment:
current bottleneck (in terms of raw speed. maybe the bottleneck is in algo, namely alpha-beta prunning)